src/components/explore/ProbeExplorer.svelte

```diff
@@ -56,19 +56,71 @@
       ? normData.filter((d) => !isEmpty(d.non_norm_histogram))
       : normData;
 
-  function smoothenData(data, accessor, level) {
-    // Interpolates percentiles values by applying a Moving Average.
-    if (!level) {
-      return data;
+
+  function getInterpPercBtnRanksForHistogram(histogram, percentiles) {
+    // Compute cumulative frequencies
+    var cumFreq = [];
+    var totalFreq = 0;
+    for (var i = 0; i < histogram.length; i++) {
+      totalFreq += histogram[i].value;
+      cumFreq.push({ bin: histogram[i].bin, cumFreq: totalFreq });
+    }
+    const normalizer = cumFreq.at(-1).cumFreq
+    if (normalizer != 1) {
+      cumFreq = cumFreq.map((item) => {return {bin: item.bin, cumFreq: item.cumFreq/normalizer}})
     }
+
+    // Find the interval where each percentile falls and interpolate
+    const percentileValues = {}
+    for (const percentile of percentiles) {
+      var targetFreq = percentile / 100;
+      if (targetFreq <= cumFreq[0].cumFreq) {
+        percentileValues[percentile] = cumFreq[0].bin;
+      }
+      if (targetFreq >= cumFreq.at(-1).cumFreq) {
+        percentileValues[percentile] = cumFreq[cumFreq.length - 1].bin;
+      }
+      for (var i = 0; i < cumFreq.length - 1; i++) {
+        if (cumFreq[i].cumFreq <= targetFreq && cumFreq[i + 1].cumFreq >= targetFreq) {
+          var x0 = cumFreq[i].cumFreq;
+          var x1 = cumFreq[i + 1].cumFreq;
+          var y0 = cumFreq[i].bin;
+          var y1 = cumFreq[i + 1].bin;
+          // Linear interpolation formula
+          var percentileValue = y0 + ((targetFreq - x0) * (y1 - y0)) / (x1 - x0);
```

Comment:
I think this works, but I'm not sure how the bin values are given. If one bucket is for values 3-7, then will `.bin` be 3, 7, or (3+7)/2=5? With exponentially increasing bucket sizes, it seems like it would make a difference. ...after more research, I think your formula is exactly right as long as the bins are given as the lowest value in each bucket, which seems like it must be the case. (Then the range of values that fall in some bucket i are [cumFreq[i].bin, cumFreq[i+1].bin) which is exactly what you're using here, so I should shut up already.)

It's a little sketchy, because the linear interpolation assumes that the values are uniformly distributed _within_ the bucket. Apparently sometimes people will use a logarithmic interpolation instead, which is for when the values are exponentially distributed within the bucket. I'm not at all sure about this stuff, but I don't think we can assume any particular distribution. And fortunately, the linear interpolation won't be very far off most other distributions. I think?

Reply:
Your assumptions are correct. I added a note about this in the tooltip that will show next to the check-box that toggles the feature.
I also think that if we need to add different types of interpolation based on something (e.g. probe metadata) we can do that quickly now that we already have the core feature implemented